xbmc/threads/Thread.cpp

```diff
@@ -243,3 +255,69 @@ void CThread::Action()
   }
 }
 
+void CThread::SetCrashProtection(IThreadCrashProtection* crashProtection)
+{
+  if (g_application.IsCurrentThread())
+    g_application.SetCrashProtection(crashProtection);
```

Comment:
Here is the call to CApplication, it is possible to change them there but to call:
`CThread::SetCrashProtection` sounds better and shows what is involved as `CApplication::SetCrashProtection`